### PR TITLE
Sort some modules into ALWAYS-BUILD in mpconfig

### LINKED ISSUE #2191: STM32: MINIMAL_BUILD flag contains modules that aren't HAL dependent
The `MINIMAL_BUILD` flag currently removes a number of modules by default that are not HAL dependent. These include OS, Struct, Storage, and potentially others. Modules that aren't dependent on a common-hal component should be sorted out and not excluded via `MINIMAL_BUILD`, which will be less confusing for new port builds and remove unnecessary restrictions on ongoing ports like the STM32.

### PR DESCRIPTION
**Pull request by hierophect**
The following modules have no unique HAL component capable of disrupting the build process of a very early stage port. 

- math
- struct
- supervisor
- time

Thus, there is never any reason to not include them, even when the MINIMAL_BUILD flag is set. This PR adds an ALWAYS_BUILD flag, which is set unless instructed otherwise, to identify these modules. 

Resolves #2191

### CODE REVIEW

File: py/circuitpy_mpconfig.mk [hunk 1]
```diff
@@ -198,7 +203,7 @@ endif
 CFLAGS += -DCIRCUITPY_PS2IO=$(CIRCUITPY_PS2IO)
 
 ifndef CIRCUITPY_RANDOM
-CIRCUITPY_RANDOM = $(CIRCUITPY_DEFAULT_BUILD)
+CIRCUITPY_RANDOM = $(CIRCUITPY_ALWAYS_BUILD)
```
Comment (tannewt) [inline py/circuitpy_mpconfig.mk]: Doesn't this rely on the OS stub at least?

File: py/circuitpy_mpconfig.mk [hunk 2]
```diff
@@ -264,12 +269,12 @@ endif
 CFLAGS += -DCIRCUITPY_UHEAP=$(CIRCUITPY_UHEAP)
 
 ifndef CIRCUITPY_USB_HID
-CIRCUITPY_USB_HID = $(CIRCUITPY_DEFAULT_BUILD)
+CIRCUITPY_USB_HID = $(CIRCUITPY_ALWAYS_BUILD)
```
Comment (tannewt) [inline py/circuitpy_mpconfig.mk]: How do these work if USB isn't available?

Comment (hierophect) [reply py/circuitpy_mpconfig.mk]: I thought these didn't interrupt compilation, but on double checking it seems I messed up my macro test and they do halt things. I'll remove them.

File: py/circuitpy_mpconfig.mk (hunk 1 above)
Comment (hierophect) [reply py/circuitpy_mpconfig.mk]: You're right, I missed that.

Comment (hierophect) [reply py/circuitpy_mpconfig.mk]: We could think about adding a true default OS stub, though, since it's so simple

**Comment by hierophect** [comment]
@tannewt I revised my test which was missing several module dependancies. USB HID and MIDI are out, but Storage and Random only rely on stub-level implementations of Microcontroller and OS (they use functions that use only CMSIS or don't suffer from being empty/returning 0). Should we consider these stubs a part of the "base build" of Circuitpython? Or should I exclude them?

**Comment by tannewt** [comment]
Merge if it's ready. :-)

**Pull request merged**